packages/metro-react-native-babel-preset/src/configs/main.js

```diff
@@ -77,7 +77,12 @@ const getPreset = (src, options) => {
   }
 
   if (hasClass) {
-    extraPlugins.push([require('@babel/plugin-transform-classes')]);
+    // If the code makes heavy use of classes, we can save a lot of bytes by
+    // avoiding injecting `@babel/runtime` helpers.
+    extraPlugins.push([
+      require('@babel/plugin-transform-classes'),
+      {loose: !options.dev},
```

Comment:
Further to my other comments: we should also strongly avoid compiling with different semantics in dev vs prod unless absolutely necessary. Diverging creates the risk of prod-only bugs and a JS environment that is harder to reason about.